main/views.py

```diff
@@ -1,57 +1,198 @@
-    data = []
-    for player in players:
-        data.append({"id": player.id, "name": player.name})
-    return JsonResponse({"players": data})
+    if request.method == "GET":
+        players = Player.objects.all()
+        data = [{"id": player.id, "name": player.name} for player in players]  # should I be doing the hanging indents here?
+        return JsonResponse({"players": data})
+    elif request.method == "POST":
+        data = json.loads(request.body)
+        player = Player.objects.create(name=data["name"])
+        return JsonResponse({
+            "id": player.id,
+            "name": player.name
+        })
+    else:
+        return HttpResponseNotAllowed(["POST", "GET"])
+
+
+def api_fields(request):
+    if request.method == "GET":
+        fields = Field.objects.all()
+        data = [{"id": field.id, "name": field.name} for field in fields]
+        return JsonResponse({
+            "fields": data
+        })
+    elif request.method == "POST":
+        data = json.loads(request.body)
+        field = Field.objects.create(name=data["name"])
+        return JsonResponse({
+            "id": field.id,
+            "name": field.name
+        })
+    else:
+        return HttpResponseNotAllowed(["POST", "GET"])
+
+
+def api_matches(request):
+    if request.method == "GET":
+        matches = Match.objects.all()
+        # where should I put the curly brackets here based on usual BIO conventions, generally should this be done this way? could be hard to read.
+        data = [{
+                "id": match.id,
+                "date_and_time": match.date_and_time,
+                "player1": match.player1.name,
+                "player2": match.player2.name,
+                "player1_score": match.player1_score,
+                "player2_score": match.player2_score,
+                "field": match.field.name
+                }
+                for match in matches
+                ]
+        return JsonResponse({
+            "matches": data
+        })
+    elif request.method == "POST":
+        data = json.loads(request.body)
+        if Player.objects.filter(id=data["player1_id"]).exists():
+            p1 = Player.objects.get(id=data["player1_id"])
+        else:
+            return HttpResponseBadRequest("Bad request: You tried to create a match with a player that doesn't exist yet.")
+
+        if Player.objects.filter(id=data["player2_id"]).exists():
+            p2 = Player.objects.get(id=data["player2_id"])
+        else:
+            return HttpResponseBadRequest("Bad request: You tried to create a match with a player that doesn't exist yet.")
+
+        if Field.objects.filter(id=data["field_id"]).exists():
+            field = Field.objects.get(id=data["field_id"])
+        else:
+            return HttpResponseBadRequest("Bad request: You tried to create a match with a player that doesn't exist yet.")
+
+        match = Match.objects.create(
+            date_and_time=datetime.datetime.now(),
```

Comment:
This works, but you could also configure this field in the model in a way so that it is automatically set to the current date at object creation, and then you wouldn't have to do it manually.

Reply:
Discussed in person: we'll update this in a new PR, if at all.